src/docfx/build/xref/XrefMap.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Docs.Build
+{
+    public class XrefMap
+    {
+        // TODO: key could be uid+moniker+locale
+        private readonly ConcurrentDictionary<string, XRefSpec> _map = new ConcurrentDictionary<string, XRefSpec>();
```

Comment:
make this a `ReadOnlyDictionary` to ensure it is not mutated after construction. Follow the pattern used in `RedirectionMap.Create`